### PR TITLE
Enable i18n

### PR DESCRIPTION
**Pull request by joseferben**

### CODE REVIEW

**Comment by joseferben** [comment]
@mabiede This is the API your proposed, what do you think?

File: src/conformist.ml [hunk 1]
```diff
@@ -1,330 +1,18 @@
...
-        match Ptime.of_rfc3339 (List.hd string) with
-        | Ok (timestamp, _, _) -> Ok timestamp
-        | Error (`RFC3339 (_, _)) -> Error msg
-      with
-      | _ -> Error msg
-    in
-    let encoder ptime = List.[ Ptime.to_rfc3339 ptime ] in
-    make name meta decoder encoder default "time" validator false
-  ;;
+module String_error : Core.ERROR = struct
```
Comment (mabiede) [inline src/conformist.ml:1]: @joseferben  when trying to pin this version, I get the following error message during the installation process

```
#    val validate :
#            'a any_field -> string list/2 -> String_error.error option
#        is not included in
#          val validate : 'a any_field -> string List.t -> error_msg option
#        File "src/core.ml", line 39, characters 4-68: Expected declaration
#        File "src/core.ml", line 344, characters 8-16: Actual declaration
#        File "src/core.ml", lines 333-337, characters 4-40:
#          Definition of type list/1
#        File "_none_", line 1:
#          Definition of type list/2
```

Comment (joseferben) [reply src/conformist.ml:1]: @mabiede Yes I couldn't get it to compile yet, just drafted the API :)

Comment (joseferben) [reply src/conformist.ml]: @mabiede Compilation errors fixed, in `test.ml` you find an example on how to apply the custom error module. What do you think?

**Comment by aronerben** [comment]
Change looks good to me, can be merged if @mabiede approves.

**Comment by mabiede** [comment]
> Change looks good to me, can be merged if @mabiede approves.

@joseferben  LGTM 😉

**Pull request merged**